src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -305,57 +305,6 @@ final class HTTPClient: Client {
 		traceWarningPackageDownload(country: country, packageId: packageId, url: url, completion: completion)
 	}
 
-	// swiftlint:disable:next cyclomatic_complexity
-	func dccRegisterPublicKey(
-		isFake: Bool = false,
-		token: String,
-		publicKey: String,
-		completion: @escaping DCCRegistrationCompletionHandler
-	) {
-		guard let request = try? URLRequest.dccPublicKeyRequest(
-			configuration: configuration,
-			token: token,
-			publicKey: publicKey,
-			headerValue: isFake ? 1 : 0
-		) else {
-			Log.error("Failed to create request")
-			completion(.failure(.urlCreationFailed))
-			return
-		}
-		session.response(for: request) { [weak self] result in
-			self?.queue.async {
-				switch result {
-				case let .success(response):
-					switch response.statusCode {
-					case 201:
-						completion(.success(()))
-					case 400:
-						completion(.failure(.badRequest))
-					case 403:
-						completion(.failure(.tokenNotAllowed))
-					case 404:
-						completion(.failure(.tokenDoesNotExist))
-					case 409:
-						completion(.failure(.tokenAlreadyAssigned))
-					case 500:
-						completion(.failure(.internalServerError))
-					default:
-						Log.error("Error in response with status code: \(String(response.statusCode))", log: .api)
-						completion(.failure(.unhandledResponse(response.statusCode)))
-					}
-				case let .failure(error):
-					if case .noNetworkConnection = error {
-						Log.error("No network connection", log: .api)
-						completion(.failure(.noNetworkConnection))
-					} else {
-						Log.error("Error in response body", log: .api, error: error)
-						completion(.failure(.defaultServerError(error)))
```

Comment:
I can't see this error handled in the new resource. Did you consider it?

Reply:
It's handled by the usual service errors now, no need for a separate case to wrap it anymore